### PR TITLE
Add premake/5.0.0-alpha14

### PR DESCRIPTION
**Pull request by Croydon**
Specify library name and version:  **premake/5.0.0-alpha14**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

____________

Based on https://github.com/bincrafters/conan-premake_installer

Fixes https://github.com/conan-io/wishlist/issues/162

### CODE REVIEW

**Comment by madebr** [comment]
Should premake use the lua library, generated by #545?

**Comment by Croydon** [comment]
> Should premake use the lua library, generated by #545?


I'm unfamiliar with how premake works internal

I'm assuming that there is a lua library included in the premake exeutable and that we would need to patch premake in order to use another lua library?

If that should be the case, that I would clearly say no

File: recipes/premake/5.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @property
+    def _platform(self):
+        return {'Windows': 'vs2017',
+                'Linux': 'gmake.unix',
+                'Macos': 'gmake.macosx'}.get(str(self.settings.os_build))
+
+    def build(self):
+        with tools.chdir(os.path.join(self._source_subfolder, 'build', self._platform)):
+            if self.settings.os_build == 'Windows':
```
Comment (madebr) [inline recipes/premake/5.x/conanfile.py]: This is only valid when `compiler == "Visual Studio"`
I think there is another build directory when using mingw.

Comment (Croydon) [reply recipes/premake/5.x/conanfile.py]: Even when handling MinGW explicitly here the build fails eventually
https://github.com/bincrafters/conan-center-index/commit/80ff869bde76fd75844357ebf1975cc8f7b9a270

For the moment I will raise an `ConanInvalidConfiguration` when trying to build with MinGW

File: recipes/premake/5.x/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from conans import ConanFile, tools
+
+
+class TestPackageConan(ConanFile):
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            self.run("premake5 --version")
```
Comment (uilianries) [inline recipes/premake/5.x/test_package/conanfile.py]:
```suggestion
            self.run("premake5 --version", run_environment=True)
```

File: recipes/premake/5.x/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/premake/5.x/conanfile.py]: Yeah, please, just add a section for mingw and we are good!

File: recipes/premake/5.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    name = "premake"
+    version = "5.0.0-alpha14"
+    topics = ("conan", "premake", "build", "build-systems")
+    description = "Describe your software project just once, using Premake's simple and easy to read syntax, and build it everywhere"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://premake.github.io/"
+    license = "BSD-3-Clause"
+    settings = "os_build", "arch_build", "compiler"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
```
Comment (uilianries) [inline recipes/premake/5.x/conanfile.py]:
```suggestion
```

Not used.

**Pull request merged**